### ISSUE TITLE
false-positive: `invalid-assignment` when using `dict.get()` with a list default

### ISSUE THREAD

**Issue by sahilmtayade** [astral-sh/ty#3949]
## Description

`ty` reports an `invalid-assignment` error when using `dict.get(key, default)` where the default is a list literal (e.g., `[0] * 12`). The inferred return type of `.get()` includes `int` as a union element, which is not assignable to the declared `list[float]` — but this appears to be a false positive because the actual runtime value will always be a `list` given the provided default.

## Reproduction

```python
data: dict = {}

result: list[float] = data.get("some_key", [0] * 12)
```

**Error:**

```
2 | result: list[float] = data.get("some_key", [0] * 12)
  |                       --------------------------------
  |                       Incompatible value of type `Unknown | int | float | ... omitted 3 union elements`
info: element `int` of union `Unknown | int | float | ... omitted 3 union elements` is not assignable to `list[int | float]`
```

## Expected Behavior

No type error should be raised. When `dict.get()` is called with a default value of `[0] * 12` (a `list[int]`), the return type should be narrowed to account for the type of the default, making the union at minimum `list[int]`-compatible. The declared type `list[float]` is assignable from `list[int]` in most contexts (or is at least not a clear error).

## Actual Behavior

`ty` infers the return type of `.get()` as a broad union (`Unknown | int | float | ...`) without narrowing based on the concrete default argument, and then flags the assignment as invalid.

## Environment

Vscode Ty Extension: 2026.60.0
Ty version: ty==0.0.54
Python == 3.11.14

## Additional Notes

This may be related to how `ty` models the return type of `dict.get()` generically without specialising on the type of the `default` argument. Ideally, `dict.get(key, default: D)` should return `V | D` where `V` is the value type, allowing accurate narrowing.
This might also highlight `ty`'s poor handling of list[int] vs list[int | float] vs list[float]

**Comment by dhruvmanila** [comment]
I'm unable to reproduce this using the provided code snippet on the [Playground](https://play.ty.dev/65a62b29-b8b8-4548-9ec4-962088a69d0b). Can you try running it on the latest ty version which is 0.0.57 and you've provided 0.0.54 ?

**Comment by carljm** [comment]
The provided reproduction also passes without error on 0.0.54, as well as on 0.0.57 and on main (playground). Closing, can reopen if more information is provided that allows reproducing the issue.
